File: 프로그래머스1단계/문자열내림차순배치.cpp
#include <string>
#include <vector>
#include <algorithm>

//sort(begin, end, greater)

using namespace std;

string solution(string s) {
    string answer = "";
    char temp;
    
    for(int i=0; i < s.size(); i++){
        if(i < s.size()-1){
            if(s[i] > s[i+1]){
                temp = s[i];
                s[i] = s[i+1];
                s[i+1] = temp;
            }
        }
    }
    
    sort(s.begin(), s.end(), greater<char>());
    answer = s;
    return answer;
}

File: 프로그래머스1단계/문자열내림차순배치.java
import java.util.*;

public class 문자열내림차순배치 {
    public String solution(String s) {
        String answer = "";
        
        String[] str = s.split("");
        Arrays.sort(str, Collections.reverseOrder());
        answer = String.join("", str);
        
        return answer;
    }
}
